### ISSUE TITLE
GitHub Actions: Automatically move a newly created issue into the Project Board

### ISSUE THREAD

**Issue by macho-catt** [hackforla/website#2306]
### Overview
As a developer, we have to ensure that our kanban board is organized for all teams so that productivity is high. For this issue, we want newly created issues to automatically be placed in the Project Board.

### Action Items

- [x] Review the notes below to understand the GitHub actions architecture we are currently moving towards
- [x] This should be a new job under the ["Issue Trigger" workflow](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/issue-trigger.yml)
    - [x] This job triggers when an issue is opened
    - [x] This job should have an if statement that triggers when the event name is "issues" and event action is "opened." 
        - if: ${{ github.event_name == 'issues' && github.event.action == 'opened'}}
    - [x] This job should run in parallel with the other action, ["Add-Missing-Labels-To-Issues"](https://github.com/hackforla/website/blob/f6df44e3200b477a753475264c945ef7d3e1b268/.github/workflows/issue-trigger.yml#L7)
      - [x] Essentially, the action should add the issue onto the Project Board AND run the action "Add-Missing-Labels-To-Issues"
  
### Pseudo-code

- [x] When an issue is opened
    - [x] Add the issue into the Project Board
    - [x] Place the issue into the "New Issue Approval" column (this action runs automatically with another GHA)

### Checks

- [x] Test in your local environment that it works
- [x] If a user creates an issue and does not add the issue to the Project Board, the GitHub action should automatically add the issue to the Project Board in the New Issue Approval Column
- [x] If a user creates an issue and adds it properly to the Project Board, the GitHub action should also move the issue to the New Issue Column in the Project Board if the user did not already do so
- [x] If the column the issue is going to has automation turned on, the GitHub action should still trigger and add the issue to the New Issue Approval column without causing any weird issues / side effects

### Resources/Instructions
Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)
[GitHub Complex Workflows doc](https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows)
[GitHub Actions Workflow Directory](https://github.com/hackforla/website/tree/gh-pages/.github/workflows)
[Events that trigger workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)
[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
[actions/github-script](https://github.com/actions/github-script)
[GitHub RESTAPI](https://docs.github.com/en/rest)

#### Architecture Notes

The idea behind the refactor is to organize our GitHub Actions so that developers can easily maintain and understand them. Currently, we want our GitHub Actions to be structured like so based on this [proposal](https://docs.google.com/spreadsheets/d/12NcZQoyGYlHlMQtJE2IM8xLYpHN75agb/edit#gid=1231634015):

- Schedules (military time)
        - Schedule Friday 0700
        - Schedule Thursday 1100
        - Schedule Daily 1100
- Linters
        - Lint SCSS
- PR Trigger
        - Add Linked Issue Labels to Pull Request
        - Add Pull Request Instructions
- Issue Trigger
        - Add Missing Labels To Issues
- WR - PR Trigger
        - WR Add Linked Issue Labels to Pull Request
        - WR Add Pull Request Instructions
- WR - Issue Trigger

Actions with the same triggers (excluding linters, which will be their own category) will live in the same github action file. Scheduled actions will live in the same file if they trigger on the same schedule (i.e. all files that trigger everyday at 11am will live in one file, while files that trigger on Friday at 7am will be on a separate file).

That said, this structure is not set in stone. If any part of it feels strange, or you have questions, feel free to bring it up with the team so we can evolve this format!

**Comment by clayton1111** [comment]
Availability: all week
ETA: 7/25

**Comment by ExperimentsInHonesty** [comment]
@clayton1111 given that the 29th is your last day, and you have not yet submitted a pr.  Are you going to submit one or will you be annotating this issue.  I am concerned if you submit a pr and then are not able to fix any changes anyone finds, that it will tie the whole thing up.

**Comment by clayton1111** [comment]
@ExperimentsInHonesty Figuring out how to test the GHA took a lot longer than I was expecting, so I have been working on creating a guide to explain how to to test GHA items. My plan is to annotate this issue since I won't have enough time to submit a PR and give reviewers enough time to fix everything!

**Comment by clayton1111** [comment]
Updates: Did not finish issue because I had difficulties figuring out how to test GHAs locally and my internship is ending. Changed focus to help create a guide explaining how to test GHAs. Link to [act library](https://github.com/nektos/act#homebrew-linuxmacos) that lets you test GHAs locally.

**Comment by lilyarj** [comment]
Availability: 3hrs
ETA: 9/20

**Comment by KazushiR** [comment]
Availability: 6hrs
ETA: 02/26/2023

**Comment by KazushiR** [comment]
1. Progress Looking into articles related to Github Actions
2. Blockers Figuring out how to use Github Actions
3. Availability Weekdays
4. ETA 03/06/2023

**Comment by KazushiR** [comment]
Progress Working on making a testing environment for my workflow
Blockers Looking at the secrets page to link my workflow to the project pages.
Availability Weekdays
ETA N/A

**Comment by KazushiR** [comment]
unassigning myself from this issue due to the wording of the direction the nature of the problem.

**Comment by jdingeman** [comment]
Reworded to make it clearer. Adding it back to the prioritized backlog.

**Comment by perlaroyerc** [comment]
Availability: 11-7 ET M-F
ETA: April 3 2023

**Comment by perlaroyerc** [comment]
Progress: Learning how to create actions and run jobs in parallel. 
Blockers: No blockers, only tons of material to read. 
Availability: 11-7 ET M-F
ETA: 4/7/23

**Comment by perlaroyerc** [comment]
Progress: Created a job that triggers when an issue is opened.
Blockers: Learning how to test this action locally.
Availability: 11-7 ET M-F
ETA: 4/21/23

**Comment by ExperimentsInHonesty** [comment]
@perlaroyerc Please either provide an update or unassign yourself from the issue and put it in the questions and review column with a comment.

**Comment by adrianang** [comment]
- Availability: MTWThSu, 6 PM - 10 PM (PST) (Weeks of 7/17, 7/24)
- ETA: I plan on finishing this issue by Monday, 7/24 at 10 PM.

**Comment by adrianang** [comment]
**Progress:** Still working on the issue; have mostly been reading through the resources on GHA and looking at PRs related to GHA/automation. 
**Blockers:** No current blockers.
**Availability:** 
 - Saturdays/Sundays: Generally from 2 PM - 8 PM PST
 - Weekdays, excluding Wednesdays: Generally from 6 PM - 10 PM PST

**ETA:** Tuesday, 8/1 by 10 PM

**Comment by adrianang** [comment]
Progress: Continuing/coming back to work on this issue; my first attempt wasn't too fruitful and I confused myself a bit, but I think I have a better idea on approaching this issue after looking at some of the other scripts and GHA issues.
Availability:
Saturdays/Sundays: Generally from 2 PM - 8 PM PST
Weekdays, excluding Wednesdays: Generally from 6 PM - 10 PM PST
ETA: Friday, 9/1 by 10 PM

**Comment by LRenDO** [comment]
Hi @adrianang! Thanks for working on this issue! If you're still working on it, please add an update. If you're not able to continue to working on it, leave us a comment to let us know.  If we don't hear from you soon, we will remove you from the issue. Thanks!

**Comment by adrianang** [comment]
Progress: I tried creating a job script for the action but didn't get too far and ended up taking a break this past month. I'll try working on it again in the upcoming week, but may have to unassign myself afterwards if I still don't have substantial progress.
Availability:
Saturdays/Sundays: Generally from 2 PM - 8 PM PST
Weekdays, excluding Wednesdays: Generally from 6 PM - 10 PM PST
ETA: Friday, 10/13 by 10 PM

**Comment by LRenDO** [comment]
Hi @adrianang! Thanks for the update! Feel free to ping me here or on Slack, if there's a blocker you want to talk through.

**Comment by adrianang** [comment]
**Progress:** 

- I added a new job to the Issue Trigger workflow that uses [Alex Page's project card automation action](https://github.com/alex-page/github-project-automation-plus) (which we also use in our Pull Request Trigger workflow) and got it to work on my local environment and cloned project board on my fork.
- I tested the checks by creating issues that were (1) not assigned to a Project and (2) assigned to the Project Board. In both cases, the newly created issues were assigned to the Project Board on the New Issue Approval column. Since the process of creating a new issue only allows the two options of either not assigning it to a project or assigning it to a project (but not the column), I believe it encompasses the last check as well (but not 100% sure)!
- I'll use the weekend to think of any edge cases I might have not accounted for and to also create the pull request for this issue.

**Blockers:**

- No blocker for the GHA; might need clarification for the last check, especially if I incorrectly interpreted (or didn't account for) different ways developers might create an issue.

**Availability:**

- Saturdays/Sundays: Generally from 2 PM - 8 PM PST
- Weekdays, excluding Wednesdays: Generally from 6 PM - 10 PM PST

**ETA:** Sunday, 10/22 by 10 PM PST
